mock/py/mockbuild/buildroot.py

```diff
@@ -497,15 +498,39 @@ def _setup_devices(self):
             kver = os.uname()[2]
             self.root_log.debug("kernel version == %s", kver)
             for i in devFiles:
+                src_path = "/" + i[2]
+                chroot_path = self.make_chroot_path(i[2])
+
+                if util.cmpKernelVer(kver, '2.6.18') >= 0 and src_path == '/dev/ptmx':
+                    continue
+
                 # create node, but only if it exist on host too
                 # except for loop devices, which only show up on the host after they are first used
-                if os.path.exists("/" + i[2]) or "loop" in i[2]:
-                    os.mknod(self.make_chroot_path(i[2]), i[0], i[1])
+                if os.path.exists(src_path) or "loop" in src_path:
+                    try:
+                        os.mknod(chroot_path, i[0], i[1])
+                    except OSError as e:
+                        # If mknod gives us a permission error, fall back to a different
+                        # strategy of using a bind mount from root to host. This won't
+                        # work for the loop devices, so just skip them in this case.
+                        if e.errno == errno.EPERM:
+                            if os.path.exists(src_path):
+                                self.mounts.add_device_bindmount(src_path)
+                            continue
+                        else:
+                            raise
+
                     # set context. (only necessary if host running selinux enabled.)
                     # fails gracefully if chcon not installed.
                     if self.selinux:
-                        util.do(["chcon", "--reference=/" + i[2], self.make_chroot_path(i[2])],
+                        util.do(["chcon", "--reference=" + src_path, chroot_path],
                                 raiseExc=0, shell=False, env=self.env)
+
+                    if src_path == '/dev/tty':
```

Comment:
Can you change this to `src_path in ['/dev/tty', '/dev/ptmx']` ?

Additionaly, can you add comment like "This is run only if not bind-mounted". ('cos it took me some time to comprehend why you took it inside of the for-loop).